google-cloud-translate/MIGRATING.md

```diff
@@ -0,0 +1,248 @@
+`Google::Cloud::Translate::V3::TranslationServiceClient` object for the
+V3 version of the service. All these classes were defined in the same gem.
+
+With the 3.0 release, the `google-cloud-translate` gem still provides factory
+methods for obtaining clients. (The method signatures will have changed. See
+[Creating Clients](#creating-clients) for details.) However, the actual client
+classes have been moved into separate gems, one per service version. The
+`Google::Cloud::Translate::V2::Api` class, along with its
+helpers and data types, is now part of the `google-cloud-translate-v2` gem.
+Similarly, the `Google::Cloud::Translate::V3::TranslationService::Client`
+class is part of the `google-cloud-translate-v3` gem. 
+
+For normal usage, you can continue to install the `google-cloud-translate` gem
+(which will bring in the versioned client gems as dependencies) and continue to
+use factory methods to create clients. However, you may alternatively choose to
+install only one of the versioned gems. For example, if you know you will use only
+`V2` of the service, you can install `google-cloud-translate-v2` by itself, and
+call `Google::Cloud::Translate::V2.new` to create V2 clients directly.
+
+### Creating Clients
+
+In older releases, to create a client object, you would use the
+`Google::Cloud::Translate.new` class method. Keyword arguments were available to
+select a service version and to configure parameters such as credentials and
+timeouts. Furthermore, you could configure default parameters using the
+`Google::Cloud::Translate.configure` method.
+
+In the 3.0 release, there are separate class methods for creating clients of the
+modern (V3) and legacy (V2) Translation services. To create a V2 client, use the
+`translation_v2_service` class method, which takes the same keyword arguments
+you would have used previously. To create a V3 (or later) client, use the
+`translation_service` class method and set options in a configuration block
+rather than passing keyword arguments.
+
+Old:
+```
+client = Google::Cloud::Translate.new credentials: "/path/to/credentials.json"
+```
+
+New (V2):
+```
+client = Google::Cloud::Translate.translation_v2_service(
+  credentials: "/path/to/credentials.json")
+```
+
+New (V3):
+```
+# Using default settings
+client1 = Google::Cloud::Translate.translation_service
+
+# Overriding default settings
+client2 = Google::Cloud::Translate.translation_service do |config|
+  config.credentials = "/path/to/credentials.json"
+end
+```
+
+### Passing Arguments
+
+In older releases, required arguments would be passed as positional method
+arguments, while most optional arguments would be passed as keyword arguments.
+
+With the 3.0 release, the V2 client interface remains the same, but in the V3
+client interface, all RPC arguments are passed as keyword arguments, regardless
+of whether they are required or optional. For example:
+
+Old:
+```
+client = Google::Cloud::Translate.new version: :v3
```

Comment:
Why specify `version: :v3` explicitly in the "Old" examples but not in the "New" examples? I think it would be better to omit it if possible to highlight the change being discussed (in this case keywords for all params).

Reply:
Updated.